projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/experiment/ExperimentalVariable.java

```diff
@@ -53,7 +54,13 @@ private ExperimentalVariable(String name, ExperimentalValue... levels) {
       throw new IllegalArgumentException("At least one variable level required.");
     }
     this.name = VariableName.create(name);
-    this.levels.addAll(List.of(levels));
+    // necessary to test unit cohesion
+    for(ExperimentalValue level : levels) {
+      Result<VariableLevel, Exception> addLevelResult = addLevel(level);//TODO this does not seem like the correct way to use this
+      if (addLevelResult.isFailure()) {
+          throw new IllegalArgumentException(addLevelResult.exception().getMessage());
+      }
+    }
```

Comment:
Why not simply check for the unit and throw the illegal argument exception?

Reply:
because that would be some code duplication. Might be better than before, though...